zsh-history-substring-search.zsh

```diff
@@ -246,7 +246,11 @@ _history-substring-search-begin() {
     # Escape and join query parts with wildcard character '*' as seperator
     # `(j:CHAR:)` join array to string with CHAR as seperator
     #
-    local search_pattern="*${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
+    if [[ -z $HISTORY_SUBSTRING_SEARCH_FUZZY_AFTER_CURSOR ]]; then
+	local search_pattern="${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
```

Comment:
Could you factor out the duplication? 😱 It's safe to always execute this line. 🤓 Next, you can conditionally prefix the `*` if this fuzzy-after-cursor option is not enabled.